android/expoview/src/main/java/host/exp/exponent/modules/ExponentKernelModule.java

```diff
@@ -112,13 +113,31 @@ public void consumeEventQueue() {
   }
 
   @ReactMethod
-  public void setSessionSecret(String sessionSecret) {
-    ExponentUrls.setSessionSecret(sessionSecret);
+  public void getSessionAsync(Promise promise) {
+    String sessionSecret = mExponentSharedPreferences.getString(ExponentSharedPreferences.EXPO_AUTH_SESSION);
+    promise.resolve(sessionSecret);
   }
 
   @ReactMethod
-  public void removeSessionSecret() {
-    ExponentUrls.setSessionSecret(null);
+  public void setSessionAsync(String sessionSecret, Promise promise) {
+    try {
+      mExponentSharedPreferences.setString(ExponentSharedPreferences.EXPO_AUTH_SESSION, sessionSecret);
```

Comment:
Won't this be the session JSON string as opposed to the secret token alone? In the iOS implementation there's all that JSON parsing that's missing here.